src/shogun/features/RandomKitchenSinksDotFeatures.h

```diff
@@ -224,7 +224,13 @@ class RandomKitchenSinksDotFeatures : public RandomMixin<DotFeatures>
 	 *
 	 * @return a random parameter vector
 	 */
-	virtual SGVector<float64_t> generate_random_parameter_vector()=0;
+	virtual void generate_random_params(SGVector<float64_t>& vec) = 0;
+	
+	/**  Get number of parameters per sample in the random_coeff matrix
```

Comment:
minor. simplify to one line
"@return number of parameters per random feature dimension"

Reply:
actually the superclass has a good comment, so this can just be removed

Reply:
Sorry, I think you meant inside the RandomFourierDotFeatures.h ?

Reply:
sorry it is not the superclass. Yes it should be the same comment as in RandomFourierDotFeatures

Reply:
Mind if I shift it from fourier.h -> sinks.h? (for both generate_random_parms and get_num_params)

Reply:
what do you mean by "shift"?

Reply:
remove from fourier, put it inside sinks

Reply:
I dont understand :D

Reply:
In RandomFourierDotFeatures.h, we have protected methods ```generate_random_params``` and ```get_num_params``` right?. I suggest removing their docs because it is already documented in RandomKitchenSinks.h.

Reply:
I see, yes. That's what I wrote above: if a superclass has a good docstring for a purely virtual method it is not always necessary to re-state that in the subclass. doxygen already place the superclass docstrings in subclasses, see #4258

Reply:
Okay, done shifting. Thanks for the details :) , resolving this..?

Reply:
I actually prefer to not resolve anything as that hides our interactions. Ppl can hide conversations if they want